docs/design/rules-api.md

```diff
@@ -0,0 +1,119 @@
+# Rules API
+
+Observatorium is featured with a multi-tenant API that enables tenants to write and read Prometheus recording and alerting rules.
+
+The API is [defined](https://github.com/observatorium/api/blob/main/rules/spec.yaml) using the [OpenAPI](https://swagger.io/specification/) specification format.
+
+The goal is to enable tenants to create, modify and access their own rules.
+
+## Usage
+
+### Create rules
+
+```
+PUT /api/v1/rules/raw
+```
+
+Set rules for a tenant.
+
+#### Example request
+
+```bash
+curl -X PUT --data-binary @alerting-rule.yaml --header "Content-Type: application/yaml" http://<observatorium-api-url>/api/metrics/v1/<tenant>/api/v1/rules/raw
+```
+
+Where:
+
+* `alerting-rule.yaml` is a YAML file containing the desired rule definition. It can contain recording rules, alerting rules or both.
+  * *Note: Each time a PUT request is made to the `/api/v1/rules/raw` endpoint, the rules contained in the request will overwrite all other rules for that tenant.*
+* `<observatorium-api-url>` is the URL where the Observatorium API is hosted.
+* `<tenant>` is the tenant name
+
+#### Example of a rule.yaml file
+
+The `rule.yaml` should be defined following the [Observatorium OpenAPI specification](https://github.com/observatorium/api/blob/main/rules/spec.yaml). The syntax is based on the Prometheus [recording](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/) and [alerting](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/) rules syntax.
+
+Example of a `rule.yaml` file containing an alerting rule:
+
+```yaml
+groups:
+  - name: test-alerting-rule
+    interval: 30s
+    rules:
+    - alert: HighRequestLatency
+      expr: job:request_latency_seconds:mean5m{job="myjob"} > 0.5
+      for: 10m
+      labels:
+        severity: page
+      annotations:
+        summary: High request latency
+```
+
+#### Example response
+
+```
+successfully updated rules file
+```
+
+| Status Code | Description                                                                                           |
+|-------------|-------------------------------------------------------------------------------------------------------|
+| 200         | Successfully listed rules.                                                                            |
```

Comment:
Do we get `Successfully listed rules.` after calling `PUT /api/v1/rules/raw` ?

Reply:
Surely `successfully updated rules file`

Reply:
oops, yes that was a copy-paste error :sweat_smile: good catch, updated it :)